src/lib/gundb/UserStorageClass.js

```diff
@@ -1593,15 +1626,22 @@ export class UserStorage {
         currentUser: this.gunuser.is.pub,
       })
 
-      //now that we use the hash of the email/mobile there's no privacy issue
+      // now that we use the hash of the email/mobile there's no privacy issue
       // if (privacy !== 'public' && indexValue !== undefined) {
       //   return indexNode.putAck(null)
       // }
 
-      const res = await indexNode.putAck(this.gunuser)
-      return res
-    } catch (e) {
-      logger.error('indexProfileField failed', e.message, e, { field })
+      return await indexNode.putAck(this.gunuser)
+    } catch (ackOrException) {
+      let exception = ackOrException
```

Comment:
Maybe move this peace of code:
```
let exception = ackOrException
      let { message } = exception

      if (ackOrException.err) {
        message = ackOrException.err
        exception = new Error(message)
      }
```
into separate function (e.g. 'handleError') and reuse it where needed

Reply:
i would do this in the promisifyGun and even started implementation but 

a) we may got an trace issues.  So let's create an errors where we've got them
b) this breaks .putAck / secretAck. interface. Before they were resolved/rejected always with the ack object. So if we'll reject with an Error tihs could break something. We have to perform an additional research for that. So let's keep the current implementation